test/functorch/test_control_flow.py

```diff
@@ -1471,44 +1726,860 @@ def fct_pointwise(x, y):
+            )[ind] @ torch.ones(2, 5, device=device)
+            result1 = fct_cmp(inp)
+            self.assertEqual(result1, expected_result)
+
+        # Chain with scan
+        def chain_fct_same_dim(inp):
+            o1 = scan(add, init=init, input=inp, dim=1, reverse=reverse)
+            o2 = scan(add, init=init, input=o1[1], dim=1, reverse=reverse)
+            return o2
+
+        torch.compiler.reset()
+        if compile_mode == "compile":
+            fct_cmp = torch.compile(chain_fct_same_dim, fullgraph=True, dynamic=False)
+        elif compile_mode == "compile_dynamic_shape":
+            fct_cmp = torch.compile(chain_fct_same_dim, fullgraph=True, dynamic=True)
+        elif compile_mode == "eager":
+            fct_cmp = torch.compile(chain_fct_same_dim, fullgraph=True, backend="eager")
+        else:
+            fct_cmp = chain_fct_same_dim
+
+        expected_result = _fake_scan(
+            add,
+            init=init,
+            input=_fake_scan(add, init=init, input=inp, dim=1, reverse=reverse)[1],
+            dim=1,
+            reverse=reverse,
+        )
+        result1 = fct_cmp(inp)
+        self.assertEqual(result1, expected_result)
+
+        # Chain with scan on different dim
+        init2 = torch.randn(1, 10, 2, device=device)
+
+        def chain_fct_different_dim(inp):
+            o1 = scan(add, init=init, input=inp, dim=1, reverse=reverse)
+            o2 = scan(add, init=init2, input=o1[1], dim=0, reverse=reverse)
+            return o2
+
+        torch.compiler.reset()
+        if compile_mode == "compile":
+            fct_cmp = torch.compile(
+                chain_fct_different_dim, fullgraph=True, dynamic=False
+            )
+        elif compile_mode == "compile_dynamic_shape":
+            fct_cmp = torch.compile(
+                chain_fct_different_dim, fullgraph=True, dynamic=True
+            )
+        elif compile_mode == "eager":
+            fct_cmp = torch.compile(
+                chain_fct_different_dim, fullgraph=True, backend="eager"
+            )
+        else:
+            fct_cmp = chain_fct_different_dim
+
+        expected_result = _fake_scan(
+            add,
+            init=init2,
+            input=_fake_scan(add, init=init, input=inp, dim=1, reverse=reverse)[1],
+            dim=0,
+            reverse=reverse,
+        )
+        result1 = fct_cmp(inp)
+        self.assertEqual(result1, expected_result)
+
+    @unittest.skipIf(not SM70OrLater, "triton")
+    @requires_cuda
+    @parametrize("reverse", [False, True])
+    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    # Skipping the combination of associative_scan and device=cpu
+    # as the current implementation of pointwise does only support CUDA device
+    @decorateIf(
+        unittest.skip,
+        lambda params: (params["device"] == torch.device("cpu")),
+    )
+    def test_generic_scan_non_pointwise(self, reverse, device):
```

Comment:
The name could be generic_associative_scan_non_pointwise, right? Since we've separated all scan tests with associative_scan tests. Can you help auditing the test names?

Reply:
I went through the test names and hopefully changed the appropriately.